handlers/user.go

```diff
@@ -1298,50 +1322,86 @@ func UserUpdateEmailPost(w http.ResponseWriter, r *http.Request) {
 		http.Redirect(w, r, "/user/settings", http.StatusSeeOther)
 		return
 	}
-	email := r.FormValue("email")
 
-	if !utils.IsValidEmail(email) {
-		session.AddFlash("Error: Invalid email format!")
+	// check if password is correct
+	formPassword := r.FormValue("password")
+
+	err = bcrypt.CompareHashAndPassword([]byte(userData.Password), []byte(formPassword))
+	if err != nil {
+		session.AddFlash("Error: Invalid email or password!")
```

Comment:
**minor:** This Flash message was copied from the login where it makes sense that we don't tell the user which of them is wrong.
However here the email can't be "wrong" it can just have the wrong format as the second flash message says.
So should this one just be something like `Error: Wrong password!`?

Reply:
writing "wrong password" flat out seems like an anti-pattern to me, changed it to "invalid credentials"